tests/subsys/audio_module/src/functional_test.c

```diff
@@ -0,0 +1,1801 @@
+
+		zassert_equal(ret, 0, "Open function did not return successfully (0): ret %d", ret);
+		zassert_equal(handle.state, AUDIO_MODULE_STATE_CONFIGURED,
+			      "Open state not AUDIO_MODULE_STATE_CONFIGURED (%d) rather %d",
+			      AUDIO_MODULE_STATE_CONFIGURED, handle.state);
+		zassert_mem_equal(&handle.name, test_inst_name, sizeof(test_inst_name),
+				  "Failed open, name should be %s, but is %s", test_inst_name,
+				  handle.name);
+		zassert_mem_equal(handle.description, &mod_description,
+				  sizeof(struct audio_module_description),
+				  "Failed open, module descriptions differ");
+		zassert_mem_equal(handle.description->functions, &mod_functions_populated,
+				  sizeof(struct audio_module_functions),
+				  "Failed open, module function pointers differ");
+		zassert_mem_equal(&handle.thread, &mod_parameters.thread,
+				  sizeof(struct audio_module_thread_configuration),
+				  "Failed open, module thread settings differ");
+		zassert_mem_equal(handle.context, &test_mod_context, sizeof(struct mod_context),
+				  "Failed open, module contexts differ");
+		zassert_equal(handle.use_tx_queue, false,
+			      "Open failed use_tx_queue is not false: %d", handle.use_tx_queue);
+		zassert_equal(handle.dest_count, 0, "Open failed dest_count is not 0: %d",
+			      handle.dest_count);
+
+		k_thread_abort(handle.thread_id);
+	}
+}
+
+ZTEST(suite_audio_module_functional, test_data_tx_fnct)
+{
+	int ret;
+	char *test_inst_name = "TEST instance 1";
+	size_t size;
+	char test_data[TEST_MOD_DATA_SIZE];
+	struct audio_data audio_data = {0};
+	struct audio_module_message *msg_rx;
+
+	test_context_set(&mod_context, &mod_config);
+
+	/* Fake internal empty data FIFO success */
+	data_fifo_init_fake.custom_fake = fake_data_fifo_init__succeeds;
+	data_fifo_pointer_first_vacant_get_fake.custom_fake =
+		fake_data_fifo_pointer_first_vacant_get__succeeds;
+	data_fifo_block_lock_fake.custom_fake = fake_data_fifo_block_lock__succeeds;
+	data_fifo_pointer_last_filled_get_fake.custom_fake =
+		fake_data_fifo_pointer_last_filled_get__succeeds;
+	data_fifo_block_free_fake.custom_fake = fake_data_fifo_block_free__succeeds;
+
+	data_fifo_deinit(&mod_fifo_rx);
+
+	data_fifo_init(&mod_fifo_rx);
+
+	memcpy(&handle.name, test_inst_name, sizeof(test_inst_name));
+	handle.description = &mod_description;
+	handle.thread.msg_rx = &mod_fifo_rx;
+	handle.thread.msg_tx = NULL;
+	handle.thread.data_slab = &mod_data_slab;
+	handle.thread.data_size = TEST_MOD_DATA_SIZE;
+	handle.state = AUDIO_MODULE_STATE_RUNNING;
+	handle.context = (struct audio_module_context *)&mod_context;
+
+	for (int i = 0; i < TEST_MOD_DATA_SIZE; i++) {
+		test_data[i] = TEST_MOD_DATA_SIZE - i;
+	}
+
+	audio_data.data = &test_data[0];
+	audio_data.data_size = TEST_MOD_DATA_SIZE;
+
+	ret = audio_module_data_tx(&handle, &audio_data, NULL);
+	zassert_equal(ret, 0, "Data TX function did not return successfully (0):: ret %d", ret);
```

Comment:
I think all the checks of `ret` can be simplified. If it fails, the line number is printed.
Also, I if i recall, the assert also prints the "ret" value, so there is no need to do it in the string.

```suggestion
	zassert_equal(ret, 0, "Did not return successfully");
```

Reply:
Some are not changed.

Reply:
In the logs I get it does not print the return value without it being in the string.

Reply:
OK. Then we can have a look at that. It is minor so I will resolve.